### PR TITLE
Update monitoring library to use new generated client

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

File: monitoring/README.rst [hunk 1]
```diff
@@ -1,78 +1,100 @@
...
-|pypi| |versions|
+- `Client Library Documentation`_
+- `Product Documentation`_
 
--  `Documentation`_
-
-.. _Documentation: https://googlecloudplatform.github.io/google-cloud-python/latest/monitoring/usage.html
+.. _Alpha: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/README.rst
+.. _Stackdriver Monitoring API: https://cloud.google.com/monitoring
+.. _Client Library Documentation: https://googlecloudplatform.github.io/google-cloud-python/stable/monitoring-usage
```
Comment (supriyagarg) [inline monitoring/README.rst]: This link is broken. Replace this with the following?
https://googlecloudplatform.github.io/google-cloud-python/latest/monitoring/usage.html

**Comment by jonparrott** [comment]
Okay, I've done a lot of work to make this branch current.

1. I re-organized everything under `monitoring_v3`. `google.cloud.monitoring` is just an alias to `monitoring_v3` (same as other packages).
2. I have enabled testing with pandas.
3. I have dropped the old handwritten docs and have imported the autogen docs. I have carried over the `query` docs.
4. I have re-generated the gapic. This should now contain the alerts feature (and more).

@supriyagarg I need you help as a few tests for query are failing and I'm not 100% sure how they should be fixed. Can you please take a look? (see test output [here](https://gist.github.com/jonparrott/1cbd6d31ac2fb2da2813ebcc3e50b640)). Feel free to push new commits to this branch to fix those and ping me if you run into issues.

It would also be great if you could give the docs a once-over. I've staged them [here](http://temp.theadora.io/google-cloud-monitoring/monitoring/index.html) for easy review.

/cc @SurferJeffAtGoogle

**Comment by supriyagarg** [comment]
I just sent a PR #5215 to merge into monitoring-autogen.
I still see some coverage issues, but having trouble seeing which lines are not covered.

File: monitoring/tests/unit/test_query.py [hunk 2]
```diff
@@ -127,14 +146,13 @@ def test_constructor_maximal(self):
...
+        self.assertEqual(query._cross_series_reducer, 0)
         self.assertEqual(query._group_by_fields, ())
 
     def test_constructor_default_end_time(self):
-        import datetime
-        import mock
+        from google.cloud.monitoring_v3 import MetricServiceClient
 
         MINUTES = 5
         NOW, T0, T1 = [
```
Comment (crwilcox) [inline monitoring/tests/unit/test_query.py]: This appears to be an old line, but it is odd that this test use multiple assignment and not three single assignments.
```
NOW = datetime.datetime(...)
T0 = datetime.datetime(...)
T1 = datetime.datetime(...)
```

Is there any rationale of building the array and extracting to the tuple?

File: monitoring/google/cloud/monitoring_v3/_dataframe.py [hunk 3]
```diff
@@ -15,6 +15,12 @@
 """Time series as :mod:`pandas` dataframes."""
 
 import itertools
+try:
+    import pandas
```
Comment (crwilcox) [inline monitoring/google/cloud/monitoring_v3/_dataframe.py]: The extras requires a min version of 17.1. Do we need to do that check here as well?

File: monitoring/README.rst [hunk 4]
```diff
@@ -1,78 +1,103 @@
...
+- `Product Documentation`_
 
-|pypi| |versions|
-
--  `Documentation`_
-
-.. _Documentation: https://googlecloudplatform.github.io/google-cloud-python/latest/monitoring/usage.html
+.. _Alpha: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/README.rst
+.. _Stackdriver Monitoring API: https://cloud.google.com/monitoring
+.. _Client Library Documentation: https://googlecloudplatform.github.io/google-cloud-python/stable/monitoring/usage.html
```
[the remark this answers is not in the document]
Comment (jonparrott) [reply monitoring/README.rst]: Fixed.

File: monitoring/README.rst [hunk 5]
```diff
@@ -1,78 +1,103 @@
...
+
+1. `Select or create a Cloud Platform project.`_
+2. `Enable billing for your project.`_
+3. `Enable the Stackdriver Monitoring API.`_
+4. `Setup Authentication.`_
+
+.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
+.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
+.. _Enable the Stackdriver Monitoring API.:  https://cloud.google.com/monitoring
+.. _Setup Authentication.: https://googlecloudplatform.github.io/google-cloud-python/stable/core/auth.html
```
[the remark this answers is not in the document]
Comment (jonparrott) [reply monitoring/README.rst]: Fixed.

File: monitoring/docs/index.rst [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+`Stackdriver Monitoring API`_: Manages your Stackdriver Monitoring data and configurations. Most projects
+must be associated with a Stackdriver account, with a few exceptions as
+noted on the individual method pages.
+
+- `Client Library Documentation`_
+- `Product Documentation`_
+
+.. _Alpha: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/README.rst
+.. _Stackdriver Monitoring API: https://cloud.google.com/monitoring
+.. _Client Library Documentation: https://googlecloudplatform.github.io/google-cloud-python/stable/monitoring/usage.html
```
[the remark this answers is not in the document]
Comment (jonparrott) [reply monitoring/docs/index.rst]: Done.

File: monitoring/google/cloud/monitoring_v3/_dataframe.py (hunk 3 above)
Comment (jonparrott) [reply monitoring/google/cloud/monitoring_v3/_dataframe.py]: I don't think so.

File: monitoring/tests/unit/test_query.py (hunk 2 above)
Comment (jonparrott) [reply monitoring/tests/unit/test_query.py]: Fixed.

**Comment by jonparrott** [comment]
Merging despite flaky system test, hopefully that's just a relatively rare 500 error.

**Pull request merged**